posix/unix.lisp

```diff
@@ -729,7 +729,8 @@ than C's printf) with format string FORMAT and arguments ARGS."
     (%readdir-r dir entry result)
     (if (null-pointer-p (mem-ref result :pointer))
         nil
-        (with-foreign-slots ((name type fileno) entry (:struct dirent))
+        #+(or sunos solaris) (foreign-string-to-list name)
```

Comment:
The `name` variable doesn't exist and neither does the `foreign-string-to-list` function.

Reply:
I just updated and this time I actually used opendir/readdir at the REPL.  I guess all my dependent quicklisp systems never even called readdir.